### PR TITLE
Publications Collection View

### PR DESCRIPTION
**Pull request by alexkb0009**
Should look like:

If gets large enough for both title + authors to fit on 1 line each, then first part of abstract should appear:

#### Responsiveness (re: column width) should work predictably.

### CODE REVIEW

File: src/encoded/static/components/browse/PublicationSearchView.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
+'use strict';
+
+import React from 'react';
+import memoize from 'memoize-one';
+import _ from 'underscore';
```
Comment (j1z0) [inline src/encoded/static/components/browse/PublicationSearchView.js]: Codacy Issue found: ['_' is defined but never used. (no-unused-vars)](https://app.codacy.com/app/4dn/fourfront/pullRequest?prid=4551745)

**Comment by j1z0** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- src/encoded/static/components/browse/PublicationSearchView.js  4
- src/encoded/static/components/browse/SearchView.js  2
         

Clones added
============
- src/encoded/static/components/browse/PublicationSearchView.js  1
- src/encoded/static/components/browse/SearchView.js  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/fourfront/fourfront/pullRequest?prid=4551745&bid=15312545)

File: src/encoded/static/components/browse/PublicationSearchView.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+import { SearchView as CommonSearchView } from '@hms-dbmi-bgm/shared-portal-components/es/components/browse/SearchView';
+import { console } from '@hms-dbmi-bgm/shared-portal-components/es/components/util';
```
Comment (j1z0) [inline src/encoded/static/components/browse/PublicationSearchView.js]: Codacy Issue found: ['console' is defined but never used. (no-unused-vars)](https://app.codacy.com/app/4dn/fourfront/pullRequest?prid=4551745)

**Comment by j1z0** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- src/encoded/static/components/browse/PublicationSearchView.js  4
- src/encoded/static/components/browse/SearchView.js  2
         

Clones added
============
- src/encoded/static/components/browse/PublicationSearchView.js  1
- src/encoded/static/components/browse/SearchView.js  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/fourfront/fourfront/pullRequest?prid=4551745&bid=15312545)

File: src/encoded/static/components/browse/PublicationSearchView.js [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+            'list'      : analytics.hrefToListName(href),
+            'position'  : rowNumber + 1
+        }, function(){
+            navigate(object.itemUtil.atId(result));
+        });
+        return false;
+    }
+
+    render(){
+        const { result, detailOpen, toggleDetailOpen, rowNumber, href, navigate = spcNavigate, width: colWidth } = this.props;
```
Comment (j1z0) [inline src/encoded/static/components/browse/PublicationSearchView.js]: Codacy Issue found: ['navigate' is assigned a value but never used. (no-unused-vars)](https://app.codacy.com/app/4dn/fourfront/pullRequest?prid=4551745)

File: src/encoded/static/components/browse/PublicationSearchView.js [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+        const { title: origTitle, "@id" : id, abstract, authors = null } = result; // We use "title" here, not "display_title" (which contains year+author, also)
```
Comment (j1z0) [inline src/encoded/static/components/browse/PublicationSearchView.js]: Codacy Issue found: ['origTitle' is assigned a value but never used. (no-unused-vars)](https://app.codacy.com/app/4dn/fourfront/pullRequest?prid=4551745)

File: src/encoded/static/components/browse/PublicationSearchView.js [hunk 5]
```diff
@@ -5,12 +5,101 @@ import memoize from 'memoize-one';
...
+        const titleLen = abstract.length;
+        const titleMaxLen = PublicationSearchResultTitle.maxCharsToShowForTitle(colWidth) * 3;
+        let title = abstract;
```
Comment (alexkb0009) [inline src/encoded/static/components/browse/PublicationSearchView.js]: Just using abstract for testing. No long titles are present in test inserts and too lazy / not-enough-justifiable-time to create some.

**Comment by j1z0** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 4
           

Complexity increasing per file
==============================
- src/encoded/static/components/browse/SearchView.js  2
- src/encoded/static/components/browse/PublicationSearchView.js  6
         

Clones added
============
- src/encoded/static/components/browse/SearchView.js  1
- src/encoded/static/components/browse/PublicationSearchView.js  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/fourfront/fourfront/pullRequest?prid=4551745&bid=15312545)

File: src/encoded/static/components/browse/PublicationSearchView.js [hunk 6]
```diff
@@ -0,0 +1,245 @@
...
+        }
+    },
+    "journal" : {
+        "widthMap" : { "sm" : 100, "md" : 120, "lg" : 150 },
+        "render" : function(result, columnDefinition, props, termTransformFxn){
+            const { journal = null } = result;
+            if (!journal) return null;
+            return (
+                <span className="value">
+                    <JournalTitle journal={journal} width={props.width} />
```
Comment (j1z0) [inline src/encoded/static/components/browse/PublicationSearchView.js]: Codacy Issue found: [Must use destructuring props assignment (react/destructuring-assignment)](https://app.codacy.com/app/4dn/fourfront/pullRequest?prid=4551745)

File: src/encoded/static/components/browse/PublicationSearchView.js [hunk 7]
```diff
@@ -0,0 +1,245 @@
...
+            'position'  : rowNumber + 1
+        }, function(){
+            navigate(object.itemUtil.atId(result));
+        });
+        return false;
+    }
+
+    render(){
+        const { result, detailOpen, toggleDetailOpen, rowNumber, href, navigate = spcNavigate, width: colWidth } = this.props;
+        const { title: origTitle, "@id" : id, abstract, authors = null } = result; // We use "title" here, not "display_title" (which contains year+author, also)
```
Comment (j1z0) [inline src/encoded/static/components/browse/PublicationSearchView.js]: Codacy Issue found: ['origTitle' is assigned a value but never used. (no-unused-vars)](https://app.codacy.com/app/4dn/fourfront/pullRequest?prid=4551745)

File: src/encoded/static/components/browse/PublicationSearchView.js [hunk 8]
```diff
@@ -0,0 +1,245 @@
...
+            'list'      : analytics.hrefToListName(href),
+            'position'  : rowNumber + 1
+        }, function(){
+            navigate(object.itemUtil.atId(result));
+        });
+        return false;
+    }
+
+    render(){
+        const { result, detailOpen, toggleDetailOpen, rowNumber, href, navigate = spcNavigate, width: colWidth } = this.props;
```
Comment (j1z0) [inline src/encoded/static/components/browse/PublicationSearchView.js]: Codacy Issue found: ['navigate' is assigned a value but never used. (no-unused-vars)](https://app.codacy.com/app/4dn/fourfront/pullRequest?prid=4551745)

File: src/encoded/static/components/browse/PublicationSearchView.js [hunk 9]
```diff
@@ -0,0 +1,245 @@
+'use strict';
+
+import React, { useMemo } from 'react';
+import memoize from 'memoize-one';
+import _ from 'underscore';
+
+import { SearchView as CommonSearchView } from '@hms-dbmi-bgm/shared-portal-components/es/components/browse/SearchView';
+import { console, analytics, object, navigate as spcNavigate, valueTransforms } from '@hms-dbmi-bgm/shared-portal-components/es/components/util';
```
Comment (j1z0) [inline src/encoded/static/components/browse/PublicationSearchView.js]: Codacy Issue found: ['console' is defined but never used. (no-unused-vars)](https://app.codacy.com/app/4dn/fourfront/pullRequest?prid=4551745)

**Comment by j1z0** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 4
           

Complexity increasing per file
==============================
- src/encoded/static/components/browse/SearchView.js  2
- src/encoded/static/components/browse/PublicationSearchView.js  6
         

Clones added
============
- src/encoded/static/components/browse/SearchView.js  1
- src/encoded/static/components/browse/PublicationSearchView.js  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/fourfront/fourfront/pullRequest?prid=4551745&bid=15312545)

File: src/encoded/static/components/browse/PublicationSearchView.js (hunk 7 above)
Comment (j1z0) [inline src/encoded/static/components/browse/PublicationSearchView.js]: Codacy Issue found: ['origTitle' is assigned a value but never used. (no-unused-vars)](https://app.codacy.com/app/4dn/fourfront/pullRequest?prid=4551745)

File: src/encoded/static/components/browse/PublicationSearchView.js (hunk 8 above)
Comment (j1z0) [inline src/encoded/static/components/browse/PublicationSearchView.js]: Codacy Issue found: ['navigate' is assigned a value but never used. (no-unused-vars)](https://app.codacy.com/app/4dn/fourfront/pullRequest?prid=4551745)

File: src/encoded/static/components/browse/PublicationSearchView.js (hunk 6 above)
Comment (j1z0) [inline src/encoded/static/components/browse/PublicationSearchView.js]: Codacy Issue found: [Must use destructuring props assignment (react/destructuring-assignment)](https://app.codacy.com/app/4dn/fourfront/pullRequest?prid=4551745)

File: src/encoded/static/components/browse/PublicationSearchView.js (hunk 9 above)
Comment (j1z0) [inline src/encoded/static/components/browse/PublicationSearchView.js]: Codacy Issue found: ['console' is defined but never used. (no-unused-vars)](https://app.codacy.com/app/4dn/fourfront/pullRequest?prid=4551745)

File: src/encoded/static/components/PageTitle.js [hunk 10]
```diff
@@ -43,7 +43,11 @@ const TITLE_PATHNAME_MAP = {
             return 'Search';
         },
         'calloutTitle' : function searchViewCalloutTitle(pathName, context, href, currentAction, schemas){
-            var thisTypeTitle = schemaTransforms.getSchemaTypeFromSearchContext(context, schemas);
+            const thisTypeTitle = (
+                schemaTransforms.getTitleForType(
+                    schemaTransforms.getSchemaTypeFromSearchContext(context), schemas
```
Comment (alexkb0009) [inline src/encoded/static/components/PageTitle.js]: We changed `getSchemaTypeFromSearchContext` in SPC to no longer prettify `type` (for utility + to match its functionality to docstring/title) and thus now need to wrap with `getTitleForType` to grab it from schemas.

**Comment by j1z0** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 4
           

Complexity increasing per file
==============================
- src/encoded/static/components/browse/PublicationSearchView.js  6
- src/encoded/static/components/browse/SearchView.js  2
         

Clones added
============
- src/encoded/static/components/browse/PublicationSearchView.js  3
- src/encoded/static/components/browse/SearchView.js  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/fourfront/fourfront/pullRequest?prid=4551745&bid=15312545)

File: src/encoded/static/components/browse/PublicationSearchView.js [hunk 11]
```diff
@@ -0,0 +1,246 @@
...
+            'position'  : rowNumber + 1
+        }, function(){
+            navigate(object.itemUtil.atId(result));
+        });
+        return false;
+    }
+
+    render(){
+        const { result, detailOpen, toggleDetailOpen, rowNumber, href, navigate = spcNavigate, width: colWidth } = this.props;
+        const { title: origTitle, "@id" : id, abstract, authors = null } = result; // We use "title" here, not "display_title" (which contains year+author, also)
```
Comment (j1z0) [inline src/encoded/static/components/browse/PublicationSearchView.js]: Codacy Issue found: ['origTitle' is assigned a value but never used. (no-unused-vars)](https://app.codacy.com/app/4dn/fourfront/pullRequest?prid=4551745)

File: src/encoded/static/components/browse/PublicationSearchView.js [hunk 12]
```diff
@@ -0,0 +1,246 @@
...
+        }
+    },
+    "journal" : {
+        "widthMap" : { "sm" : 100, "md" : 120, "lg" : 150 },
+        "render" : function(result, columnDefinition, props, termTransformFxn){
+            const { journal = null } = result;
+            if (!journal) return null;
+            return (
+                <span className="value">
+                    <JournalTitle journal={journal} width={props.width} />
```
Comment (j1z0) [inline src/encoded/static/components/browse/PublicationSearchView.js]: Codacy Issue found: [Must use destructuring props assignment (react/destructuring-assignment)](https://app.codacy.com/app/4dn/fourfront/pullRequest?prid=4551745)

File: src/encoded/static/components/browse/PublicationSearchView.js [hunk 13]
```diff
@@ -0,0 +1,246 @@
...
+            'list'      : analytics.hrefToListName(href),
+            'position'  : rowNumber + 1
+        }, function(){
+            navigate(object.itemUtil.atId(result));
+        });
+        return false;
+    }
+
+    render(){
+        const { result, detailOpen, toggleDetailOpen, rowNumber, href, navigate = spcNavigate, width: colWidth } = this.props;
```
Comment (j1z0) [inline src/encoded/static/components/browse/PublicationSearchView.js]: Codacy Issue found: ['navigate' is assigned a value but never used. (no-unused-vars)](https://app.codacy.com/app/4dn/fourfront/pullRequest?prid=4551745)

File: src/encoded/static/components/browse/PublicationSearchView.js [hunk 14]
```diff
@@ -0,0 +1,246 @@
+'use strict';
+
+import React, { useMemo } from 'react';
+import memoize from 'memoize-one';
+import _ from 'underscore';
+
+import { SearchView as CommonSearchView } from '@hms-dbmi-bgm/shared-portal-components/es/components/browse/SearchView';
+import { console, analytics, object, navigate as spcNavigate, valueTransforms } from '@hms-dbmi-bgm/shared-portal-components/es/components/util';
```
Comment (j1z0) [inline src/encoded/static/components/browse/PublicationSearchView.js]: Codacy Issue found: ['console' is defined but never used. (no-unused-vars)](https://app.codacy.com/app/4dn/fourfront/pullRequest?prid=4551745)

**Comment by j1z0** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 5
           

Complexity increasing per file
==============================
- src/encoded/static/components/browse/SearchView.js  2
- src/encoded/static/components/browse/PublicationSearchView.js  8
         

Clones added
============
- src/encoded/static/components/browse/SearchView.js  1
- src/encoded/static/components/browse/PublicationSearchView.js  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/fourfront/fourfront/pullRequest?prid=4551745&bid=15312545)

File: src/encoded/static/components/browse/PublicationSearchView.js [hunk 15]
```diff
@@ -0,0 +1,263 @@
...
+        }
+    },
+    "journal" : {
+        "widthMap" : { "sm" : 100, "md" : 120, "lg" : 150 },
+        "render" : function(result, columnDefinition, props, termTransformFxn){
+            const { journal = null } = result;
+            if (!journal) return null;
+            return (
+                <span className="value">
+                    <JournalTitle journal={journal} width={props.width} />
```
Comment (j1z0) [inline src/encoded/static/components/browse/PublicationSearchView.js]: Codacy Issue found: [Must use destructuring props assignment (react/destructuring-assignment)](https://app.codacy.com/app/4dn/fourfront/pullRequest?prid=4551745)

File: src/encoded/static/components/browse/PublicationSearchView.js [hunk 16]
```diff
@@ -0,0 +1,263 @@
...
+                authorsToKeep.map(function(authorStr, i){
+                    return <span className="author-name" key={authorStr}>{ authorStr }</span>;
+                })
+            }
+            { authorsAreTrimmed ? <span className="author-name"><em>et al.</em></span> : null }
+        </div>
+    );
+});
+
+const AbstractBlock = React.memo(function AbstractBlock(props){
```
Comment (j1z0) [inline src/encoded/static/components/browse/PublicationSearchView.js]: Codacy Issue found: ['AbstractBlock' is assigned a value but never used. (no-unused-vars)](https://app.codacy.com/app/4dn/fourfront/pullRequest?prid=4551745)

File: src/encoded/static/components/browse/PublicationSearchView.js [hunk 17]
```diff
@@ -0,0 +1,263 @@
...
+            'list'      : analytics.hrefToListName(href),
+            'position'  : rowNumber + 1
+        }, function(){
+            navigate(object.itemUtil.atId(result));
+        });
+        return false;
+    }
+
+    render(){
+        const { result, detailOpen, toggleDetailOpen, rowNumber, href, navigate = spcNavigate, width: colWidth } = this.props;
```
Comment (j1z0) [inline src/encoded/static/components/browse/PublicationSearchView.js]: Codacy Issue found: ['navigate' is assigned a value but never used. (no-unused-vars)](https://app.codacy.com/app/4dn/fourfront/pullRequest?prid=4551745)

File: src/encoded/static/components/browse/PublicationSearchView.js [hunk 18]
```diff
@@ -0,0 +1,263 @@
...
+                    return <span className="author-name" key={authorStr}>{ authorStr }</span>;
+                })
+            }
+            { authorsAreTrimmed ? <span className="author-name"><em>et al.</em></span> : null }
+        </div>
+    );
+});
+
+const AbstractBlock = React.memo(function AbstractBlock(props){
+    const { abstract = null } = props;
```
Comment (j1z0) [inline src/encoded/static/components/browse/PublicationSearchView.js]: Codacy Issue found: ['abstract' is assigned a value but never used. (no-unused-vars)](https://app.codacy.com/app/4dn/fourfront/pullRequest?prid=4551745)

File: src/encoded/static/components/browse/PublicationSearchView.js [hunk 19]
```diff
@@ -0,0 +1,263 @@
+'use strict';
+
+import React, { useMemo } from 'react';
+import memoize from 'memoize-one';
+import _ from 'underscore';
+
+import { SearchView as CommonSearchView } from '@hms-dbmi-bgm/shared-portal-components/es/components/browse/SearchView';
+import { console, analytics, object, navigate as spcNavigate, valueTransforms } from '@hms-dbmi-bgm/shared-portal-components/es/components/util';
```
Comment (j1z0) [inline src/encoded/static/components/browse/PublicationSearchView.js]: Codacy Issue found: ['console' is defined but never used. (no-unused-vars)](https://app.codacy.com/app/4dn/fourfront/pullRequest?prid=4551745)

File: src/encoded/schemas/publication.json [hunk 20]
```diff
@@ -171,11 +171,16 @@
...
+            "title" : "Year",
+            "notes": "We use date_published here but only render out year."
+        },
         "journal": {
             "title": "Journal"
         },
         "date_published": {
-            "title": "Publication Date"
+            "title": "Publication Date",
+            "default_hidden" : true
```
Comment (alexkb0009) [inline src/encoded/schemas/publication.json]: Show month + year. Maybe could remove this completely and/or add other columns; idk.

File: src/encoded/static/components/app.js [hunk 21]
```diff
@@ -1177,20 +1178,16 @@ export default class App extends React.PureComponent {
...
                     <link rel="preconnect" href="https://unpkg.com" />
                     <link rel="preconnect" href="https://fonts.googleapis.com/" />
                     <link rel="preconnect" href="//www.google-analytics.com" />
-                    <link rel="stylesheet" href="https://unpkg.com/rc-tabs@9.6.0/dist/rc-tabs.min.css" />
+                    <link rel="stylesheet" href="https://unpkg.com/rc-tabs@9.6.0/dist/rc-tabs.min.css" type="text/css" />
                     <SEO.CurrentContext {...{ context, hrefParts, baseDomain }} />
-                    <link href="https://fonts.googleapis.com/css?family=Mada:200,300,400,500,600,700,900|Yrsa|Source+Code+Pro:300,400,500,600" rel="stylesheet"/>
-                    <script async type="application/javascript" src={"/static/build/bundle.js?build=" + (lastCSSBuildTime || 0)} charSet="utf-8" />
+                    <link href="https://fonts.googleapis.com/css?family=Mada:200,300,400,500,600,700,900|Yrsa|Source+Code+Pro:300,400,500,600" rel="stylesheet" type="text/css"/>
+                    <script defer type="application/javascript" src={"/static/build/bundle.js?build=" + (lastCSSBuildTime || 0)} charSet="utf-8" />
```
Comment (alexkb0009) [inline src/encoded/static/components/app.js]: Appears to fix flash of unstyled content (FOUC) glitch on current versions of FireFox.

File: src/encoded/static/components/browse/PublicationSearchView.js [hunk 22]
```diff
@@ -0,0 +1,260 @@
...
+                    { valueTransforms.decorateNumberWithCommas(numSets) }
+                </span>
+            );
+        }
+    }
+});
+
+/**
+ * This is memoized into a couple of different functions - 1 per column (common colWidth between all cells in same col)
+ * Make sure to keep this in sync with CSS rules (line height, etc.)
```
Comment (alexkb0009) [inline src/encoded/static/components/browse/PublicationSearchView.js]: There's more "accurate" ways of measuring how many lines a text block takes up but not quite performant or simple. Using "6px = upper bound of char width" rough estimate (+ padding, etc) isn't accurate but is simple/performant.

Comment (alexkb0009) [reply src/encoded/static/components/browse/PublicationSearchView.js]: Alternatively could use monospace font :-P (jk)

**Comment by j1z0** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           

Complexity increasing per file
==============================
- src/encoded/static/components/browse/SearchView.js  2
- src/encoded/static/components/browse/PublicationSearchView.js  8
         

Clones added
============
- src/encoded/static/components/browse/SearchView.js  1
- src/encoded/static/components/browse/PublicationSearchView.js  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/fourfront/fourfront/pullRequest?prid=4551745&bid=15312545)

File: src/encoded/static/components/browse/PublicationSearchView.js [hunk 23]
```diff
@@ -0,0 +1,260 @@
...
+        }
+    },
+    "journal" : {
+        "widthMap" : { "sm" : 100, "md" : 120, "lg" : 150 },
+        "render" : function(result, columnDefinition, props, termTransformFxn){
+            const { journal = null } = result;
+            if (!journal) return null;
+            return (
+                <span className="value">
+                    <JournalTitle journal={journal} width={props.width} />
```
Comment (j1z0) [inline src/encoded/static/components/browse/PublicationSearchView.js]: Codacy Issue found: [Must use destructuring props assignment (react/destructuring-assignment)](https://app.codacy.com/app/4dn/fourfront/pullRequest?prid=4551745)

File: src/encoded/static/components/browse/PublicationSearchView.js [hunk 24]
```diff
@@ -0,0 +1,260 @@
...
+            'list'      : analytics.hrefToListName(href),
+            'position'  : rowNumber + 1
+        }, function(){
+            navigate(object.itemUtil.atId(result));
+        });
+        return false;
+    }
+
+    render(){
+        const { result, detailOpen, toggleDetailOpen, rowNumber, href, navigate = spcNavigate, width: colWidth } = this.props;
```
Comment (j1z0) [inline src/encoded/static/components/browse/PublicationSearchView.js]: Codacy Issue found: ['navigate' is assigned a value but never used. (no-unused-vars)](https://app.codacy.com/app/4dn/fourfront/pullRequest?prid=4551745)

File: src/encoded/static/components/browse/PublicationSearchView.js [hunk 25]
```diff
@@ -0,0 +1,260 @@
+'use strict';
+
+import React, { useMemo } from 'react';
+import memoize from 'memoize-one';
+import _ from 'underscore';
+
+import { SearchView as CommonSearchView } from '@hms-dbmi-bgm/shared-portal-components/es/components/browse/SearchView';
+import { console, analytics, object, navigate as spcNavigate, valueTransforms } from '@hms-dbmi-bgm/shared-portal-components/es/components/util';
```
Comment (j1z0) [inline src/encoded/static/components/browse/PublicationSearchView.js]: Codacy Issue found: ['console' is defined but never used. (no-unused-vars)](https://app.codacy.com/app/4dn/fourfront/pullRequest?prid=4551745)

File: src/encoded/static/components/app.js (hunk 21 above)
Comment (Bianca-Morris) [reply src/encoded/static/components/app.js]: Noticing that we're using rel="preconnect" here. That was unsupported by Safari and IE last time I checked, IIRC. I think I remember hearing something about trying to push users towards Chrome & Firefox for cgap; assuming that's the case here, too?

File: src/encoded/schemas/publication.json (hunk 20 above)
Comment (Bianca-Morris) [reply src/encoded/schemas/publication.json]: Wait, if you have the date_published, can't you parse that on the front end to get the year? Why does this 'nonexistent_year_field' need to be in the schema?

Comment (alexkb0009) [reply src/encoded/schemas/publication.json]: Is just a dummy field under `columns` and isn't a real field in DB (or anywhere else in backend) and just used on frontend as dummy field on "column" schema. Which realizing presents an issue when click "sort" .. thx for the catch!

Yeah agreed. Originally wanted to preserve `date_published` as its own column figuring that it'd have much more fine-grained info incl. day+month, but it seems to usually only have month so seems is less useful than thought originally.

File: src/encoded/static/components/app.js (hunk 21 above)
Comment (alexkb0009) [reply src/encoded/static/components/app.js]: Yep for most part. Added the preconnect in recently mostly based on [this article](https://developers.google.com/web/fundamentals/performance/optimizing-content-efficiency/loading-third-party-javascript#use_resource_hints_to_reduce_connection_setup_time); haven't tested w/ IE or Safari so not really sure what happens when unsupported but hoping it degrades gracefully or is ignored by them.

**Comment by j1z0** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           

Complexity increasing per file
==============================
- src/encoded/static/components/browse/SearchView.js  2
- src/encoded/static/components/browse/PublicationSearchView.js  8
         

Clones added
============
- src/encoded/static/components/browse/SearchView.js  1
- src/encoded/static/components/browse/PublicationSearchView.js  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/fourfront/fourfront/pullRequest?prid=4551745&bid=15312545)

File: src/encoded/static/components/browse/PublicationSearchView.js [hunk 26]
```diff
@@ -0,0 +1,260 @@
...
+        }
+    },
+    "journal" : {
+        "widthMap" : { "sm" : 100, "md" : 120, "lg" : 150 },
+        "render" : function(result, columnDefinition, props, termTransformFxn){
+            const { journal = null } = result;
+            if (!journal) return null;
+            return (
+                <span className="value">
+                    <JournalTitle journal={journal} width={props.width} />
```
Comment (j1z0) [inline src/encoded/static/components/browse/PublicationSearchView.js]: Codacy Issue found: [Must use destructuring props assignment (react/destructuring-assignment)](https://app.codacy.com/app/4dn/fourfront/pullRequest?prid=4551745)

File: src/encoded/static/components/browse/PublicationSearchView.js [hunk 27]
```diff
@@ -0,0 +1,260 @@
...
+            'list'      : analytics.hrefToListName(href),
+            'position'  : rowNumber + 1
+        }, function(){
+            navigate(object.itemUtil.atId(result));
+        });
+        return false;
+    }
+
+    render(){
+        const { result, detailOpen, toggleDetailOpen, rowNumber, href, navigate = spcNavigate, width: colWidth } = this.props;
```
Comment (j1z0) [inline src/encoded/static/components/browse/PublicationSearchView.js]: Codacy Issue found: ['navigate' is assigned a value but never used. (no-unused-vars)](https://app.codacy.com/app/4dn/fourfront/pullRequest?prid=4551745)

File: src/encoded/static/components/browse/PublicationSearchView.js (hunk 25 above)
Comment (j1z0) [inline src/encoded/static/components/browse/PublicationSearchView.js]: Codacy Issue found: ['console' is defined but never used. (no-unused-vars)](https://app.codacy.com/app/4dn/fourfront/pullRequest?prid=4551745)

**Comment by j1z0** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           

Complexity increasing per file
==============================
- src/encoded/static/components/browse/SearchView.js  2
- src/encoded/static/components/browse/PublicationSearchView.js  11
         

Clones added
============
- src/encoded/static/components/browse/SearchView.js  1
- src/encoded/static/components/browse/PublicationSearchView.js  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/fourfront/fourfront/pullRequest?prid=4551745&bid=15312545)

File: src/encoded/static/components/browse/PublicationSearchView.js [hunk 28]
```diff
@@ -0,0 +1,278 @@
...
+        }
+    },
+    "journal" : {
+        "widthMap" : { "sm" : 100, "md" : 120, "lg" : 150 },
+        "render" : function(result, columnDefinition, props, termTransformFxn){
+            const { journal = null } = result;
+            if (!journal) return null;
+            return (
+                <span className="value">
+                    <JournalTitle journal={journal} width={props.width} />
```
Comment (j1z0) [inline src/encoded/static/components/browse/PublicationSearchView.js]: Codacy Issue found: [Must use destructuring props assignment (react/destructuring-assignment)](https://app.codacy.com/app/4dn/fourfront/pullRequest?prid=4551745)

File: src/encoded/static/components/browse/PublicationSearchView.js [hunk 29]
```diff
@@ -0,0 +1,278 @@
+'use strict';
+
+import React, { useMemo } from 'react';
+import memoize from 'memoize-one';
+import _ from 'underscore';
+
+import { SearchView as CommonSearchView } from '@hms-dbmi-bgm/shared-portal-components/es/components/browse/SearchView';
+import { console, analytics, object, navigate as spcNavigate, valueTransforms } from '@hms-dbmi-bgm/shared-portal-components/es/components/util';
```
Comment (j1z0) [inline src/encoded/static/components/browse/PublicationSearchView.js]: Codacy Issue found: ['console' is defined but never used. (no-unused-vars)](https://app.codacy.com/app/4dn/fourfront/pullRequest?prid=4551745)

File: src/encoded/static/components/browse/PublicationSearchView.js [hunk 30]
```diff
@@ -0,0 +1,278 @@
...
+            'list'      : analytics.hrefToListName(href),
+            'position'  : rowNumber + 1
+        }, function(){
+            navigate(object.itemUtil.atId(result));
+        });
+        return false;
+    }
+
+    render(){
+        const { result, detailOpen, toggleDetailOpen, rowNumber, href, navigate = spcNavigate, width: colWidth, showAbstractBlock } = this.props;
```
Comment (j1z0) [inline src/encoded/static/components/browse/PublicationSearchView.js]: Codacy Issue found: ['navigate' is assigned a value but never used. (no-unused-vars)](https://app.codacy.com/app/4dn/fourfront/pullRequest?prid=4551745)

**Pull request merged**